GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -474,13 +475,41 @@ List<TreeNode> SearchTree(string text, IEnumerable<TreeNode> nodes)
                     {
                         if (branch.FullPath.IndexOf(text, StringComparison.InvariantCultureIgnoreCase) != -1)
                         {
+                            if (!isFullMatch)
+                            {
+                                if (string.Compare(text, branch.FullPath, StringComparison.InvariantCultureIgnoreCase) == 0)
+                                {
+                                    isFullMatch = true;
+                                    matchIndex = ret.Count;
+                                }
+                                else if (matchIndex == -1 && branch.FullPath.Contains('/') &&
+                                    string.Compare(text, branch.FullPath.Split('/').Last(), StringComparison.InvariantCultureIgnoreCase) == 0)
+                                {
+                                    matchIndex = ret.Count;
+                                }
+                            }
+
                             AddTreeNodeToSearchResult(ret, n);
                         }
                     }
                     else
                     {
                         if (n.Text.IndexOf(text, StringComparison.InvariantCultureIgnoreCase) != -1)
                         {
+                            if (!isFullMatch)
```

Comment:
This is duplication of code.
Please use a local function (or maybe a variable for branch/n.Text)

Reply:
Moved it to a local function